src/user/user.controller.ts

```diff
@@ -1,93 +1,74 @@
-import { Body, Controller, Delete, Get, Param, Patch, Post, Render, Request, UseGuards } from '@nestjs/common';
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    HttpCode,
+    Param,
+    Patch,
+    Render,
+    Request,
+    UseGuards
+} from '@nestjs/common';
 import { ApiTags } from '@nestjs/swagger';
 import { ObjectId } from 'mongoose';
 import { Roles } from '../auth/roles/roles.decorator';
 import { RolesGuard } from '../auth/roles/roles.guard';
 import { JwtAuthGuard } from '../auth/strategies/jwt/jwt-auth.guard';
+import { returnUser } from './dto/return-user.dto';
 import { UpdateUserDto } from './dto/update-user.dto';
-import { User } from './entities/user.entity';
 import { Role } from './enums/role.enum';
+import { VerifyJWTGuard } from './guards/mailVerify-jwt.guard';
 import { UserService } from './user.service';
 
 @ApiTags('user')
 @Controller('user')
 export class UserController {
-  constructor(private readonly userService: UserService) { }
+    constructor(private readonly userService: UserService) {}
 
-  @Get()
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async findAll(): Promise<any> {
-    return await this.userService.findAll();
-  }
+    @Get()
+    @UseGuards(JwtAuthGuard, RolesGuard)
+    @Roles(Role.Admin)
+    async findAll(): Promise<returnUser[]> {
+        return await this.userService.findAll();
+    }
 
-  @Get('/verify')
-  @UseGuards(JwtAuthGuard, RolesGuard)
-  @Roles(Role.Admin)
-  async getVerifyByUsername(@Body() userData: { userId: ObjectId }): Promise<any> {
-    return await this.userService.findVerify(userData.userId)
-  }
+    @Get('/verify')
+    @UseGuards(VerifyJWTGuard)
+    @Render('MailVerify')
+    async verifyMail(@Request() req): Promise<returnUser> {
+        const user = await this.userService.veryfiyUser(req.user);
+        return await this.userService.transformToReturn(user);
+    }
 
-  @Get("/verify/:code")
-  @Render("MailVerify")
-  async verifyMail(@Param('code') code: string): Promise<any> {
-    let result = await this.userService.veryfiyUser(code)
-    return
+    @Get('/profile')
+    @UseGuards(JwtAuthGuard)
+    async getProfile(@Request() req): Promise<returnUser> {
+        const user = await this.userService.findOneById(req.user.userId);
+        return await this.userService.transformToReturn(user);
+    }
 
-  }
+    @Get('/get-verify')
+    @UseGuards(JwtAuthGuard)
+    async regenerateVerify(@Request() req): Promise<void> {
+        const userData = await this.userService.parseJWTtOUsable(req.user);
+        await this.userService.createVerification(userData);
+    }
 
-  @Get('/profile')
-  @UseGuards(JwtAuthGuard)
-  getProfile(@Request() req): Promise<any> {
-    return req.user
-  }
+    @Patch('/:id')
```

Comment:
With this you can patch any user you want? Shouldn't we only allow to patch yourself. However admins are allowed to do all?